extension/src/plots/model/quickPick.ts

```diff
@@ -99,9 +100,54 @@ export const pickCustomPlotType = (): Thenable<CustomPlotType | undefined> => {
 const getTypeColumnLikes = (columns: Column[], columnType: ColumnType) =>
   columns
     .filter(({ type }) => type === columnType)
-    .map(({ label, path }) => ({ label, path }))
+    .map(({ path, label }) => ({ label, path }))
 
-export const pickMetricAndParam = async (columns: Column[]) => {
+const getCustomPlotIds = (
+  customPlotVals: CustomPlotsOrderValue[],
+  customPlotType: CustomPlotType
+) => {
+  const plotIds = new Set()
+
+  for (const { type, metric, param } of customPlotVals) {
+    if (type === customPlotType) {
+      plotIds.add(getCustomPlotId(metric, param))
+    }
+  }
+
+  return plotIds
+}
+
+const filterColumnsBasedOffCustomPlotValues = (
```

Comment:
[I] I would avoid combining the logic here. Have two different branches for pickMetric vs pickMetricAndParam.

All pickMetric needs to do is filter the full list against what is already selected (easy, understandable).

pickMetricAndParam is more involved.

My smol brain cannot cope with this